components/chatUI/ChatInput.tsx

```diff
@@ -0,0 +1,96 @@
+import { SendIcon } from "lucide-react";
+import { FormEvent, useRef, useState } from "react";
+import { MdAttachFile } from "react-icons/md";
+import { VscStopCircle } from "react-icons/vsc";
+
+interface ChatInputProps {
+  input: string;
+  handleInputChange: (
+    e:
+      | React.ChangeEvent<HTMLInputElement>
+      | React.ChangeEvent<HTMLTextAreaElement>
```

Comment:
Make it simpler, just use `(text: string) => void` and let the caller extract and pass the string